lib/ansible/parsing/mod_args.py

```diff
@@ -286,8 +290,16 @@ def parse(self):
 
         # module: <stuff> is the more new-style invocation
 
-        # walk the input dictionary to see we recognize a module name
-        for (item, value) in iteritems(self._task_ds):
+        # filter out task attributes so we're only querying unrecognized keys as actions/modules
+        non_task_ds = dict((k, v) for k, v in iteritems(self._task_ds) if (k not in self._task_attrs) and (not k.startswith('with_')))
```

Comment:
This line concerns me from a performance stand point. I haven't tested it at actual run time, but comprehensions like this were a source of issue in the variable manager and in the strategy.  It's not immediately apparent how we might improve it though, and this could be an over optimization on my part.

Reply:
Yeah, it's a set lookup and a string front-end comparison, so not too bad, and I also don't see how you could get it significantly faster without already knowing the complete set of `with_` stuff...